iphone/cli/hooks/frameworks.js

```diff
@@ -808,15 +821,20 @@ class FrameworkInfo {
 		}
 		this._path = path;
 
-		if (type !== FRAMEWORK_TYPE_STATIC && type !== FRAMEWORK_TYPE_DYNAMIC) {
-			throw new TypeError('Framework type needs to be either static or dynamic');
-		}
-		this._type = type;
+		if (!this._path.endsWith('.xcframework')) {
+			if (type !== FRAMEWORK_TYPE_STATIC && type !== FRAMEWORK_TYPE_DYNAMIC) {
+				throw new TypeError('Framework type needs to be either static or dynamic');
+			}
+			this._type = type;
 
-		if (!(architectures instanceof Set)) {
-			throw new TypeError('Framework architectures must be a set of valid architectures');
+			if (!(architectures instanceof Set)) {
+				throw new TypeError('Framework architectures must be a set of valid architectures');
+			}
+			this._architectures = architectures;
+		} else {
+			this._type = FRAMEWORK_TYPE_DYNAMIC; // FIXME: Determine if this is dynamic. We shoudl be able to look under the folders and see somehow...
+			this._architectures = new Set(); // just ignore arches?
```

Comment:
`.xcframework` packages contain a `Info.plist` we could parse to get a list of all included libraries. This can then be used to determine whether the binaries are dynamic or static. I'll implement this and then update the PR!

Reply:
@janvennemann Parsing Info.plist would be great. We need it in [_build.js](https://github.com/appcelerator/titanium_mobile/blob/eca0416a255b1cac57c6526d257860225465b0cd/iphone/cli/commands/_build.js#L2230) and [_buildModule.js](https://github.com/appcelerator/titanium_mobile/blob/193bbc3b5d8b80fc1b6f8cf4fb1dedbf12a64986/iphone/cli/commands/_buildModule.js#L631)